format_print.c

```diff
@@ -140,7 +140,10 @@ static void print_time(int t)
 	stack[p++] = s / 10 + '0';
 	stack[p++] = ':';
 	stack[p++] = m % 10 + '0';
-	stack[p++] = m / 10 + '0';
+	if (m / 10 || h || time_show_leading_zero)
+		stack[p++] = m / 10 + '0';
+	else
+		stack[p++] = ' ';
```

Comment:
Good catch, I don't use `show_remaining_time` and didn't think to test the combination of the two flags.

Come to think of it, does it make sense to pad the time with a space at all? I was originally thinking that users would want the durations to always take up the same amount of space, but that already won't be the case for durations of more than 1 hour, and anyway if a constant width is required it can be set with a format string.